### PR TITLE
Bump nodejs in CI to 16

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
Helps us remove some dependencies and use modern features without hassle

## Motivation and Context
Thanks @wkloucek and @xoxys for the fast provisioning of https://github.com/owncloud-ci/nodejs/pull/40

### CODE REVIEW

**Comment by pascalwengerter** [comment]
@fschade if we get this merged before the performance branch we can remove `core-js` (again)

**Review by kulmann — APPROVED** [review]
🚀

**Review by phil-davis — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -109,6 +109,6 @@
     "xml-js": "^1.6.11"
   },
   "engines": {
-    "node": ">=10 <=15"
+    "node": ">=10 <=16"
```
Comment (fschade) [inline package.json:112]: we should test version 10, 11, 12, 13 and 14 from which version on we support building web. I'm also fine to keep it and bring back core-js as a babel step what i did to make my performance pr buildable in ci with node 14

Comment (fschade) [reply package.json:112]: @kulmann tested node 10 and it worked, so i'm fine to bump node for ci and keep version 10 in package.json

**Review by fschade — APPROVED** [review]
👍 LGTM

**Comment by fschade** [comment]
> @fschade if we get this merged before the performance branch we can remove `core-js` (again)

we should keep core-js for now, and deprecate <= 14. Then remove it

**Comment by pascalwengerter** [comment]
> > @fschade if we get this merged before the performance branch we can remove `core-js` (again)
> 
> we should keep core-js for now, and deprecate <= 14. Then remove it

Let's do that, but in a separate PR? Think we discussed that before - who's building on Node <14 anyways?

**Comment by fschade** [comment]
> > > @fschade if we get this merged before the performance branch we can remove `core-js` (again)
> > 
> > 
> > we should keep core-js for now, and deprecate <= 14. Then remove it
> 
> Let's do that, but in a separate PR? Think we discussed that before - who's building on Node <14 anyways?
already tackled here https://github.com/owncloud/web/blob/e468d999357fd064ccb771214d80b41c63db0c71/.babelrc#L8 no need in this pr

**Pull request merged**